editor/project_converter_3_to_4.cpp

```diff
@@ -334,17 +337,21 @@ bool ProjectConverter3To4::convert() {
 	// Check file by file.
 	for (int i = 0; i < collected_files.size(); i++) {
 		String file_name = collected_files[i];
-		Vector<String> lines;
+		Vector<SourceLine> source_lines;
 		uint32_t ignored_lines = 0;
 		{
 			Ref<FileAccess> file = FileAccess::open(file_name, FileAccess::READ);
 			ERR_CONTINUE_MSG(file.is_null(), vformat("Unable to read content of \"%s\".", file_name));
 			while (!file->eof_reached()) {
 				String line = file->get_line();
-				lines.append(line);
+
+				SourceLine source_line;
+				source_line.line = line;
+				source_line.is_comment = reg_container.gdscript_comment.search_all(line).size() > 0 || reg_container.csharp_comment.search_all(line).size() > 0;
```

Comment:
Since both filters are applied to all files, we should make sure no `#` lines are typically found in C#, or `//` lines in GDScript (that one I'm fairly confident), and neither of the two in the other types of files handled by the converter (.tscn, .tres, project.godot).

Reply:
Yes. I already did some testing and it looked good.
AFAIK, `#` or `//` are only used in GDScript and C#.
But I will do more tests today.

Reply:
I continued testing and converting and found no problems.
Lines of all the files typically start with one of the following: any letter (A-Z), `;`, `"`, `[`, `]`, `}`

One interesting note: You could actually create global properties or keyinputs named `#` or `//` (`project.godot`).
![image](https://user-images.githubusercontent.com/66004280/222532067-65793000-084f-4b8a-a6d6-985b79babce3.png)
If someone actually named a property starting with a `#` or `//`, they will be ignored. But I think this is good anyway, because these are chosen by the developer and should not be influenced by the converter anyway.

Reply:
Thanks for testing!